sdcm/cluster.py

```diff
@@ -363,6 +365,7 @@ def init(self) -> None:
         # Start task threads after ssh is up, otherwise the dense ssh attempts from task
         # threads will make SCT builder to be blocked by sshguard of gce instance.
         self.wait_ssh_up(verbose=True)
+        self.wait_for_cloud_init()
```

Comment:
I wouldn't add it to base class `init` - Azure does it differently (so this will be duplication) and as you mentioned, docker/baremetal and possibly others might have problem with that.
I propose to add it to `AWSNode` and `GCENode` `init` methods only

Reply:
Agree. It is not applicable to docker/k8s.

So, need to make a `raise NotImplementedError()` implementation by default and then redefine it per backend.